odoo/sql_db.py

```diff
@@ -100,6 +101,64 @@ def check(f, self, *args, **kwargs):
         raise psycopg2.OperationalError('Unable to use a closed cursor.')
     return f(self, *args, **kwargs)
 
+class Savepoint:
+    """ Reifies an active breakpoint, allows :meth:`BaseModel.savepoint` users
+    to internally rollback the savepoint (as many times as they want) without
+    having to implement their own savepointing, or triggering exceptions.
+
+    While the savepoint name is created during (and accessible after)
+    initialisation, the actual savepoint is only defined in ``__enter__``, and
+    only valid for the context-managed span. The savepoint is automatically
+    released ("committed") on context exit, unless an exception was raised in
+    which case it's rolled back first.
```

Comment:
The word "unless" is not appropriate in this sentence IMO.  The savepoint is always released on context exit.  In case an exception was raised in context, it is rolled back before being released.

Reply:
True, in fact ensuring it's released in all cases was one of the goals (even if it apparently does nothing for the 64-savepoints-per-transaction limit).